c/src/ml-api-service-extension.c

```diff
@@ -1324,3 +1461,70 @@ ml_service_destroy (ml_service_h handle)
 
   return _ml_service_destroy_internal ((ml_service_s *) handle);
 }
+
+/**
+ * @brief Request the service related to machine-learning.
+ */
+int
+ml_service_request (ml_service_h handle, const char *config_path,
+    void *data, size_t data_len)
+{
+  ml_service_s *mls = (ml_service_s *) handle;
+  int status = ML_ERROR_NONE;
+
+  check_feature_state (ML_FEATURE_SERVICE);
+
+  if (!_ml_service_handle_is_valid (mls))
+    _ml_error_report_return (ML_ERROR_INVALID_PARAMETER,
+        "The parameter, 'handle' (ml_service_h), is invalid. It should be a valid ml_service_h instance, which is usually created by ml_service_new().");
+
+  if (!STR_IS_VALID (config_path) ||
+      !g_file_test (config_path, (G_FILE_TEST_EXISTS | G_FILE_TEST_IS_REGULAR)))
+    _ml_error_report_return (ML_ERROR_INVALID_PARAMETER,
+        "The parameter, config_path, is invalid. It should be a valid string.");
+
+  if (!data)
+    _ml_error_report_return (ML_ERROR_INVALID_PARAMETER,
+        "The parameter, data (ml_tensors_data_h), is NULL. It should be a valid ml_tensor_data_h instance, which is usually created by ml_tensors_data_create().");
+
+  switch (mls->type) {
+    case ML_SERVICE_TYPE_EXTENSION:
+      break;
+    case ML_SERVICE_TYPE_REMOTE:
+    {
```

Comment:
How about checking the type ```mls->type == ML_SERVICE_TYPE_REMOTE``` when entering a function instead of the Switch statement?

Reply:
As ml-service-extension was additionally implemented, there was no need to change it. PTAL.